### PR TITLE
Added Login and Register API

### PR DESCRIPTION
**Pull request by tylerm390**
Added login and register with email and password. The following was tested in postman:

Login with email:
- Logging in with an invalid email address (Response => "Invalid Email")
- Logging in with a valid email address that is not in the database (Response => "Invalid Credentials")
- Logging in with a valid email address but wrong password (Response => "Invalid Credentials")
- Logging in with a valid email address and valid password (Response => User ID)

Register with email:
- Registering with an invalid email address (Response => "Invalid Email")
- Registering with an email address already in use (Response => "Email in Use")
- Registering with a valid email address and valid password (Response => "User Create")

In addition, missing email and password fields in the input JSON are detected for both Login and Register with email.

Some formatting changes were also made to how I was importing certain functions.

### CODE REVIEW

**Review by gblake31 — APPROVED** [review]
I want to talk about naming choices for the response json, but I'm going to approve this now so we begin implementing the front-end part. The code isn't going to do any harm as is.

File: functions/src/index.ts [hunk 1]
```diff
@@ -43,12 +43,61 @@ app.use(
...
+  getAuth,
+  signInWithEmailAndPassword,
+  createUserWithEmailAndPassword,
+} from 'firebase/auth'
+
+const auth = getAuth()
+
+// Login user with email
+// Expects input containing email and password
+app.post('/loginWithEmail', (req: Request, res: Response) => {
```
Comment (cameroncuster) [inline functions/src/index.ts:80]: I think we should use doxygen to document the API. It's super easy to use and is pretty standard (you'll notice they did this for the climbing project)

looks like this

```
/**
 * This is a comment.
 *
 * @param name The name of the parameter.
 * @return The return value of the function.
 */
int my_function(const char* name) {
  // Do something.
  return 0;
}
```

**Review by cameroncuster — CHANGES_REQUESTED** [review]
LGTM, but IMO we should add the docs

File: functions/src/index.ts [hunk 2]
```diff
@@ -43,12 +43,89 @@ app.use(
...
+ * @req JSON containing "email" and "password" fields
+ * @res Stores the status of the request and a json containing either
+ *      the "userID" field if the request was valid, the "error" field
+ *      if the API crashed, or the "general" field if the request was
+ *      invalid. The following may be stored in the general field:
+ *          - Invalid Credentials: Email or Password was incorrect
+ *          - Invalid Email: The provided email is improperly formatted
+ *          - Missing Email: No email field was provided
+ *          - Missing Password: No password field was provided
+ */
```
Comment (cameroncuster) [inline functions/src/index.ts:79]: let's use doxygen https://www.doxygen.nl/manual/starting.html

**Review by cameroncuster — CHANGES_REQUESTED** [review]
Code is looking good, but we should document with doxygen and also set up tests for each of the API functions (this will make our system more robust)

We also need to configre the action to run our test suite (I can set this up)

I'm happy to work with you to help knock some of this out since you've already done a lot.

**Review by cameroncuster — APPROVED** [review]
Good point about not wanting this stuff to block Blake, approving.

**Pull request merged**